src/notifications/notificationmanager.cpp

```diff
@@ -103,6 +105,9 @@ NotificationManager::NotificationManager(QObject *parent) :
     databaseCommitTimer.setSingleShot(true);
     connect(&databaseCommitTimer, SIGNAL(timeout()), this, SLOT(commit()));
 
+    expirationTimer.setSingleShot(true);
+    connect(&expirationTimer, SIGNAL(timeout()), this, SLOT(expire()));
+
```

Comment:
Do you mean that the timer will not fire until after the device next wakes, or that it will be lost altogether?